freezeyt_blog/app.py

```diff
@@ -1,28 +1,42 @@
 import mistune
 from flask import Flask, url_for, abort, render_template
 from pathlib import Path
+from urllib.parse import urlparse
 
 from pygments import highlight
 from pygments.lexers import get_lexer_by_name
 from pygments.formatters import HtmlFormatter
 
 app = Flask(__name__)
+print(app.config)
 
 BASE_PATH = Path(__file__).parent
 CONTENT_PATH = BASE_PATH / 'content/'
 
 ARTICLES_PATH = CONTENT_PATH / 'articles/'
-STATIC_PATH = CONTENT_PATH / 'images/'
+IMG_PATH = 'static/images/'
 
-class HighlightRenderer(mistune.Renderer):
+app.config['IMAGES_PATH'] = ['static/images']
+
+class BlogRenderer(mistune.Renderer):
     def block_code(self, code, lang):
         if not lang:
-            return '\n<pre><code>%s</code></pre>\n' % \
-                mistune.escape(code)
+            return '\n<pre><code>%s</code></pre>\n' %mistune.escape(code)
+
         lexer = get_lexer_by_name(lang, stripall=True)
         formatter = HtmlFormatter()
         return highlight(code, lexer, formatter)
 
+    def image(self, src, alt="", title=None):
+        src = urlparse(src)
+        if not src.scheme or src.netloc:
+            filename = Path(src.path).name
+            src = f"{ url_for('static.images', filename=filename) }"
```

Comment:
První argument pro `url_for` je jméno routy – funkce za`@app.route`, tedy u nás `index` a `post`. Flask samotný definuje routu `static` pro statické soubory, ale `static.images` ne.
Napadají mě 2 způsoby, jako to vyřešit:
- Použít `url_for('static', filename='images/' + filename)`, a obrázky přesunout do adresáře `static` (kde je Flask očekává, podobně jako šablony očekává v `templates/`)
- Nadefinovat novou routu `article_image` (s podtržítkem), která přečte a daný obrázek (jako bajty, ne text, tedy  [`open(..., mode='rb')`](https://docs.python.org/3.8/library/functions.html#open)) a vrátí ho.
  Protože je to obrázek a ne HTML, doporučuju vrátit [`Response(nactene_bajty, mimetype='image/png')`](https://flask.palletsprojects.com/en/1.1.x/api/?highlight=response#flask.Response).

Druhý způsob mi přijde zajímavější :)

Reply:
Ok zkusím tedy druhý způsob, ale co mimetype když mi někdo uloží formát .jpg ?